src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/markdown/util/TableTagV2Replacer.java

```diff
@@ -0,0 +1,115 @@
+import com.github.onsdigital.babbage.content.client.ContentReadException;
+import com.github.onsdigital.babbage.content.client.ContentResponse;
+import com.github.onsdigital.babbage.error.ResourceNotFoundException;
+import com.github.onsdigital.babbage.logging.Log;
+import com.github.onsdigital.babbage.template.TemplateService;
+import com.github.onsdigital.babbage.util.http.ClientConfiguration;
+import com.github.onsdigital.babbage.util.http.PooledHttpClient;
+import com.github.onsdigital.babbage.util.json.JsonUtil;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpStatus;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Defines the format of the custom markdown tags for charts and defines how to replace them.
+ */
+public class TableTagV2Replacer extends TagReplacementStrategy {
+
+    private static final Pattern pattern = Pattern.compile("<ons-table-v2\\spath=\"([-A-Za-z0-9+&@#/%?=~_|!:,.;()*$]+)\"?\\s?/>");
+    private static final String RENDERER_HOST = Configuration.TABLE_RENDERER.getHost();
+    private static final String RENDERER_PATH = Configuration.TABLE_RENDERER.getHtmlPath();
+    static final PooledHttpClient HTTP_CLIENT = new PooledHttpClient(RENDERER_HOST, createHttpConfiguration());
+
+    private final String template;
+    private final ContentClient contentClient;
+    private final TemplateService templateService;
+    private final PooledHttpClient httpClient;
+
+    public TableTagV2Replacer(String path, String template) {
+        this(path, template, ContentClient.getInstance(), TemplateService.getInstance(), HTTP_CLIENT);
+    }
+
+    public TableTagV2Replacer(String path, String template, ContentClient contentClient, TemplateService templateService, PooledHttpClient httpClient) {
+        super(path);
+        this.template = template;
+        this.contentClient = contentClient;
+        this.templateService = templateService;
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Gets the pattern that this strategy is applied to.
+     *
+     * @return
+     */
+    @Override
+    public Pattern getPattern() {
+        return pattern;
+    }
+
+    /**
+     * The function that generates the replacement text for each match.
+     *
+     * @param matcher
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public String replace(Matcher matcher) throws IOException {
+
+        String tagPath = matcher.group(1);
+        String figureUri = resolveFigureUri(this.getPath(), Paths.get(tagPath)) + ".json";
+
+        try {
+            ContentResponse contentResponse = contentClient.getResource(figureUri);
```

Comment:
Do you need to close this?

Reply:
I was about to get all embarrased about not closing a response, but actually ContentResponse contains a bye array - the actual response has been read and closed by the method returning a ContentResponse.

Reply:
👍